pylint/checkers/variables.py

```diff
@@ -227,25 +231,26 @@ def _fix_dot_imports(
             if not isinstance(stmt, (nodes.ImportFrom, nodes.Import)):
                 continue
             for imports in stmt.names:
-                second_name = None
                 import_module_name = imports[0]
-                if import_module_name == "*":
+                second_name = (
                     # In case of wildcard imports,
                     # pick the name from inside the imported module.
-                    second_name = name
-                else:
-                    name_matches_dotted_import = False
-                    if (
-                        import_module_name.startswith(name)
-                        and import_module_name.find(".") > -1
-                    ):
-                        name_matches_dotted_import = True
-
-                    if name_matches_dotted_import or name in imports:
-                        # Most likely something like 'xml.etree',
-                        # which will appear in the .locals as 'xml'.
-                        # Only pick the name if it wasn't consumed.
-                        second_name = import_module_name
+                    name
+                    if import_module_name == "*"
+                    else
+                    # Most likely something like 'xml.etree',
+                    # which will appear in the .locals as 'xml'.
+                    # Only pick the name if it wasn't consumed.
+                    (
+                        import_module_name
+                        if (
+                            import_module_name.startswith(name)
+                            and import_module_name.find(".") > -1
+                        )
+                        or name in imports
+                        else None
+                    )
+                )
```

Comment:
I would prefer to avoid adding multi-line ternaries. I find this refactor worse than the original form. Some of the shorter ones are at best a wash.